src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,32 @@
+package lotto.domain;
+
+import lotto.domain.generator.LottoGenerator;
+import lotto.domain.number.LottoNumber;
+
+import java.util.*;
+
+public class Lotto {
+    private final HashSet<LottoNumber> lottoNumbers = new HashSet<>();
+
+    public Lotto(LottoGenerator lottoGenerator) {
+        this.lottoNumbers.addAll(lottoGenerator.generate());
+    }
+
+    public int findMatchCount(WinningLotto winningLotto) {
+        return (int) lottoNumbers.stream()
+                .filter(lottoNumber -> winningLotto.winningLotto.contains(lottoNumber))
```

Comment:
```suggestion
                .filter(lottoNumber -> winningLotto.contains(lottoNumber))
```

옆 친구와만 놀아라는 디미터 법칙 위반이라 위와 같이 구현하는 것은 어떨까?